sign_api/src/crypto/payload.rs

```diff
@@ -0,0 +1,238 @@
+use {
+    anyhow::Result,
+    base64::{prelude::BASE64_STANDARD, Engine},
+    chacha20poly1305::{
+        aead::{Aead, KeyInit, OsRng, Payload},
+        AeadCore, ChaCha20Poly1305, Nonce,
+    },
+};
+
+// https://specs.walletconnect.com/2.0/specs/clients/core/crypto/
+// crypto-envelopes
+const TYPE_0: u8 = 0;
+const TYPE_1: u8 = 1;
+const TYPE_INDEX: usize = 0;
+const TYPE_LENGTH: usize = 1;
+const IV_LENGTH: usize = 12;
```

Comment:
I'd probably name it a bit more explicitly, sth like `INIT_VEC_LEN` (`IV` looks like Roman digit 4).

NB: Everywhere `iv` is used, I'd probably have it as `InitVecFoo`, `INIT_VEC`, `init_vec` etc.